cmd/deploy.go

```diff
@@ -99,45 +99,37 @@ var deployCmd = &cobra.Command{
 		}
 
 		// create docker network or use existing one
-		if err = c.Runtime.CreateNet(ctx); err != nil {
+		if err = c.GlobalRuntime().CreateNet(ctx); err != nil {
 			return err
 		}
 
-		numNodes := uint(len(c.Nodes))
-		numLinks := uint(len(c.Links))
-		nodesMaxWorkers := maxWorkers
-		linksMaxWorkers := maxWorkers
+		nodeWorkers := uint(len(c.Nodes))
+		linkWorkers := uint(len(c.Links))
 
-		if maxWorkers == 0 {
-			nodesMaxWorkers = numNodes
-			linksMaxWorkers = numLinks
+		if maxWorkers > 0 && maxWorkers < nodeWorkers {
+			nodeWorkers = maxWorkers
 		}
 
-		if nodesMaxWorkers > numNodes {
-			nodesMaxWorkers = numNodes
-		}
-		if linksMaxWorkers > numLinks {
-			linksMaxWorkers = numLinks
+		if maxWorkers > 0 && maxWorkers < linkWorkers {
+			linkWorkers = maxWorkers
 		}
 
-		c.CreateNodes(ctx, nodesMaxWorkers)
-		c.CreateLinks(ctx, linksMaxWorkers, false)
-
-		// generate graph of the lab topology
-		if graph {
-			if err = c.GenerateGraph(topo); err != nil {
-				log.Error(err)
+		// Serializing ignite workers due to busy device error
+		for _, node := range c.Nodes {
+			if node.GetRuntime().GetName() == runtime.IgniteRuntime {
```

Comment:
I wonder, we import the whole runtime package because of a single constant. Just wondering, to make a cleaner interface separation, wouldn't it be better to have the number of workers to come from some runtime method?

Reply:
For this, I thought that in a future PR we can create two groups of nodes, the ones using `ignite` and the others.
Create the nodes with `ignite` runtime sequentially then the others with the configured workers numbers.

Reply:
good idea, let me capture this in the following issue https://github.com/srl-labs/containerlab/issues/493

Reply:
Does this matter that much? Even when you use `wg` containers still get created sequentially by `runc` (afair). 
There's not much of a difference in terms of time, maybe seconds.

Reply:
the time will grow with the number of nodes you have in the topo
for a basic 8 node lab the sequential mode takes 14sec, whereas the concurrent takes 7sec

if we can make things faster without investing much in it, I don't see why we shouldn't

Reply:
yeah, fair enough.